src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs

```diff
@@ -54,27 +54,21 @@ public override TestResult Execute(TestExecutionContext context)
 
             try
             {
-                BeforeTest(context);
-
-                context.CurrentResult = innerCommand.Execute(context);
-            }
-            catch (Exception ex)
-            {
-#if !NETSTANDARD1_3 && !NETSTANDARD1_6
-                if (ex is ThreadAbortException)
-                    Thread.ResetAbort();
-#endif
-                context.CurrentResult.RecordException(ex);
+                context.ExecuteWithExceptionHandling(c =>
+                {
+                    BeforeTest?.Invoke(c);
```

Comment:
I haven't pulled the code to look at this myself, but why do you check BeforeTest for being null, but not AfterTest?

Reply:
Oh, never mind. `ExecuteWithExceptionHandling` handles nulls.